### PR TITLE
Init front jhipster lite

### LINKED ISSUE #295: JHipster Lite internal: init front

### PR DESCRIPTION
**Pull request by nonomoho**
- fix #295

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  collectCoverage: true,
+  collectCoverageFrom: [
+    'src/main/webapp/**/*.{js,ts,vue}',
+    '!src/main/webapp/**/*.component.ts',
+    '!src/main/webapp/app/main.ts',
+    '!src/main/webapp/app/router/*.ts',
+    '!src/i18n.ts',
+    '!**/*.d.ts',
+    '!**/*AsyncModule.ts',
```
Comment (nonomoho) [inline jest.config.js:16]: Remove unused things

**Comment by pascalgrimaud** [comment]
@nonomoho @seraphinandrieux : ping when it's ready for review

**Comment by nonomoho** [comment]
It's ok, we reviewed it!

File: package.json [hunk 2]
```diff
@@ -14,17 +14,42 @@
   },
   "license": "Apache-2.0",
   "scripts": {
+    "build": "vue-tsc --noEmit && vite build --emptyOutDir",
+    "dev": "vite",
+    "format": "prettier . --write",
```
Comment (pascalgrimaud) [inline package.json:19]: is it the same thing than prettier:format ?

File: package.json [hunk 3]
```diff
@@ -14,17 +14,42 @@
...
+    "lint": "eslint --ext .js,.ts,.vue --ignore-path .gitignore --fix src/main/webapp src/test/javascript",
     "prepare": "husky install",
     "prettier:check": "prettier --check \"{,src/**/}*.{md,json,yml,html,js,ts,tsx,css,scss,vue,java,xml}\"",
-    "prettier:format": "prettier --write \"{,src/**/}*.{md,json,yml,html,js,ts,tsx,css,scss,vue,java,xml}\""
+    "prettier:format": "prettier --write \"{,src/**/}*.{md,json,yml,html,js,ts,tsx,css,scss,vue,java,xml}\"",
+    "preview": "vite preview",
+    "test": "jest src/test/javascript/spec"
+  },
+  "dependencies": {
+    "vue": "^3.2.25"
```
Comment (pascalgrimaud) [inline package.json:28]: possible to use fixed version ?

File: src/main/webapp/app/common/primary/app/App.html [hunk 4]
```diff
@@ -0,0 +1 @@
+<div id="app">{{ jhipsterLite }}</div>
```
Comment (pascalgrimaud) [inline src/main/webapp/app/common/primary/app/App.html:1]: little bit strange to have this file: `App.html`
Is it intended ? should we include it directly in `App.vue` file ?

File: src/test/javascript/spec/common/primary/app/App.spec.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { shallowMount } from '@vue/test-utils';
+import { VueWrapper } from '@vue/test-utils';
```
Comment (pascalgrimaud) [inline src/test/javascript/spec/common/primary/app/App.spec.ts:2]: maybe we could merge these 2 imports in 1 ?

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Comment by pascalgrimaud** [comment]
another comment: we should include jest-sonar-reporter, otherwise, we won't be at 100% coverage in SonarCloud

File: package.json (hunk 2 above)
Comment (pascalgrimaud) [reply package.json:19]: not important, can be changed later if needed

File: src/main/webapp/app/common/primary/app/App.html (hunk 4 above)
Comment (pascalgrimaud) [reply src/main/webapp/app/common/primary/app/App.html:1]: not important, can be changed later if needed

**Review by pascalgrimaud — APPROVED** [review]

**Comment by seraphinandrieux** [comment]
$250 bounty claim for https://github.com/jhipster/jhipster-lite/issues/295 
Link to open collective: https://opencollective.com/generator-jhipster/expenses/64192

**Comment by pascalgrimaud** [comment]
@nonomoho : approved
@seraphinandrieux : approved

thanks for your hard work

**Pull request merged**